### ISSUE TITLE
Link to common search terms

### ISSUE THREAD

**Issue by TheRealMystic** [d4rken-org/permission-pilot#24]
This isn't a bug, but the average user is most likely to use these terms when searching for permissions:

1. Internet
2. SMS
3. Microphone

Unfortunately, the system uses a different set of terms for these permissions, and as a result a search for these (and others) terms yields a blank result.

Can something be done about it to 'intelligently' show permissions that the user is trying to search:

[Screenshot_2022-07-17-20-36-38-75_ce34de6323cfaeb7a2880d42987990b1](https://user-images.githubusercontent.com/54448474/179404679-ea6c0047-1b04-4221-89d2-a9e45ab29950.jpg)

[Screenshot_2022-07-17-20-37-03-24_ce34de6323cfaeb7a2880d42987990b1](https://user-images.githubusercontent.com/54448474/179404694-ab37df00-1ae8-482f-96c6-504795f9bfe5.jpg)

[Screenshot_2022-07-17-20-39-10-19_ce34de6323cfaeb7a2880d42987990b1](https://user-images.githubusercontent.com/54448474/179404700-a7d1f793-429e-41b8-8e5a-9fb1cde71222.jpg)

**Comment by d4rken** [comment]
I've started to abstract the permissions a bit already and created a collection, see:

https://github.com/d4rken/mystic-permissions/blob/f8087a1187c21a2f7592c1f8ace64babe2b982c9/app/src/main/java/eu/darken/myperm/permissions/core/AndroidPermissions.kt#L6-L15

we could theoretically also add keywords for search and check them

https://github.com/d4rken/mystic-permissions/blob/f8087a1187c21a2f7592c1f8ace64babe2b982c9/app/src/main/java/eu/darken/myperm/permissions/ui/list/PermissionsFragmentVM.kt#L39-L54

I'm not sure how complex this will get though.
Theoretically we'd need to make them translatable too right? For users typing in other languages.

How many common terms would there be?

P.S.: If possible don't make all screenshots display by removing the `!` infront, maybe only the first. keeps it a bit easier to read. :wink:

**Comment by TheMysticS** [comment]
> I've started to abstract the permissions a bit already and created a collection, see:
> 
> https://github.com/d4rken/mystic-permissions/blob/f8087a1187c21a2f7592c1f8ace64babe2b982c9/app/src/main/java/eu/darken/myperm/permissions/core/AndroidPermissions.kt#L6-L15
> 
> we could theoretically also add keywords for search and check them
> 
> https://github.com/d4rken/mystic-permissions/blob/f8087a1187c21a2f7592c1f8ace64babe2b982c9/app/src/main/java/eu/darken/myperm/permissions/ui/list/PermissionsFragmentVM.kt#L39-L54
> 
> I'm not sure how complex this will get though. Theoretically we'd need to make them translatable too right? For users typing in other languages.
> 
> How many common terms would there be?
> 
Yes, it can get quite complex. Can you provide for just the most important/ frequently used terms only?

> P.S.: If possible don't make all screenshots display by removing the `!` infront, maybe only the first. keeps it a bit easier to read. 😉
Sure, I will do that.

**Comment by d4rken** [comment]
> Can you provide for just the most important/ frequently used terms only?

Can you provide :smirk: ?

**Comment by TheMysticS** [comment]
> > Can you provide for just the most important/ frequently used terms only?
> 
> Can you provide 😏 ?

Sure. I'll be more than happy to help you in whatever limited capacity I can.

**Comment by vertigo220** [comment]
Here's some terms that would likely be common but don't come up with any results:

- internet
- gps
- wifi (yet it's listed under connectivity, so perhaps searching not just the permission names but their descriptions would be useful)
- network
- texting (tho "text" shows the messaging permission, so if the user doesn't keep typing they'll get what they need, and sms/mms is probably more likely than texting for a user savvy enough to be using this app)
- mic/microphone
